registry/registry_test.go

```diff
@@ -0,0 +1,317 @@
+
+	runtimeclient "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+var authInfo = runtimeclient.BasicAuth("foo", "bar")
+
+func TestNewClient(t *testing.T) {
+	swaggerClient := client.New(runtimeclient.New("foobar:30002", "/api",
+		[]string{"http"}), strfmt.Default)
+	authInfo := runtimeclient.BasicAuth("foo", "bar")
+	c := NewClient(swaggerClient, authInfo)
+
+	require.NotNil(t, c)
+	assert.NotNil(t, c.AuthInfo)
+	assert.NotNil(t, c.Client)
+
+	assert.Equal(t, swaggerClient, c.Client)
+}
+
+func TestRESTClient_NewRegistry(t *testing.T) {
+	req := &model.Registry{
+		Credential: &model.RegistryCredential{},
+		Insecure:   true,
+		Name:       "testregistry",
+		Type:       "harbor",
+		URL:        "http://test.reg",
+	}
+	ctx := context.Background()
+
+	p := &mocks.MockClientService{}
+	p.On("PostRegistries",
+		&products.PostRegistriesParams{
+			Registry: req,
+			Context:  ctx,
+		},
+		mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc"),
+	).Return(&products.PostRegistriesCreated{}, nil)
+	p.On("GetRegistries",
+		&products.GetRegistriesParams{
+			Name:    &req.Name,
+			Context: ctx,
+		}, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc"),
+	).Return(
+		&products.GetRegistriesOK{
+			Payload: []*model.Registry{{
+				CreationTime: "",
+				Credential:   nil,
+				Description:  "",
+				ID:           10,
+				Insecure:     req.Insecure,
+				Name:         req.Name,
+				Status:       "",
+				Type:         req.Type,
+				UpdateTime:   "",
+				URL:          req.URL,
+			}},
+		}, nil)
+
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+
+	r, err := cl.NewRegistry(ctx, req.Name, req.Type,
+		req.URL, &model.RegistryCredential{}, true)
+
+	p.AssertExpectations(t)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(10), r.ID)
+	assert.Equal(t, req.Name, r.Name)
+	assert.Equal(t, req.URL, r.URL)
+	assert.Equal(t, req.Type, r.Type)
+	assert.Equal(t, req.Insecure, r.Insecure)
+}
+
+func TestRESTClient_NewRegistry_ErrOnPOST(t *testing.T) {
+	req := &model.Registry{
+		Credential: &model.RegistryCredential{},
+		Insecure:   true,
+		Name:       "testregistry",
+		Type:       "harbor",
+		URL:        "http://test.reg",
+	}
+	ctx := context.Background()
+
+	p := &mocks.MockClientService{}
+	p.On("PostRegistries",
+		&products.PostRegistriesParams{
+			Registry: req,
+			Context:  ctx,
+		},
+		mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc"),
+	).Return(nil, &runtime.APIError{
+		OperationName: "",
+		Response:      nil,
+		Code:          400,
+	})
+
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+
+	r, err := cl.NewRegistry(ctx, req.Name, req.Type,
+		req.URL, &model.RegistryCredential{}, true)
+
+	p.AssertExpectations(t)
+	assert.Nil(t, r)
+	if assert.Error(t, err) {
+		assert.Equal(t, "illegal format of provided ID value", err.Error())
```

Comment:
Maybe compare this to the error type containing this message instead:
```suggestion
		assert.Equal(t, &ErrRegistryIllegalIDFormatMsg{}, err)
```